src/components/markdown-renderer/hooks/dom-purifier-node-preprocessor.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import type { Document } from 'domhandler'
+import render from 'dom-serializer'
+import DOMPurify from 'dompurify'
+import { parseDocument } from 'htmlparser2'
+
+
+const customTags = ['app-linemarker', 'app-katex', 'app-gist', 'app-youtube', 'app-vimeo', 'app-asciinema']
```

Comment:
Maybe this list should be not be static but build by used plugins?

Reply:
Well. I general yes.. But we don't have a system that merges the markdown it plugins, preprocessors and replacers and splits them into plugins. We could do that. But not here.